src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof string): GroupsMap<T> {
+  const result = {};
+
+  // eslint-disable-next-line no-restricted-syntax
```

Comment:
let's remove eslint disable and only change `for of` to `.reduce()`

Reply:
Done.